### ISSUE TITLE
[BUG] Songs are grouped by album author, not album name.

### ISSUE THREAD

**Issue by k0nstruct** [FoedusProgramme/Gramophone#298]
**Describe the bug**

So you downloaded an .m4a album containing song n°1, 2, 3 and 4, made by authors A and B.

-Song 1 is made by artist A. It will be on the artist's page and album as intended.

-Song 2 is also made by artist A. It will be in the same album. So far so good.

-Song 3 is made by artist A with artist B. In the app, it will be isolated in a duplicate album with the same name. This album, for the app, it completely separate, made by a _different band_ : artist 1 and 2 together. It will only have song 3 in it.

-Song 4 is made by artist B alone. It will be put on their page, with all their other _solo_ songs. The song will also be isolated in a separate album.

Song 3 and 4 are not in the right album, but in a duplicate, made for the specific author (or combination of authors) in each song.

**TL;DR : the app doesn't handle well songs with multiple artists.**

**To Reproduce**
1. Download an album where at least one song deviates from the rest by having a different matchup of artists.
2. Open the app.
3. Find a deviant song from the album.
4. Hit the three dots and then "see album"
5. It's on its own (or with other deviant songs with the same match-up).

Inversely, if you hit "see album" for a non-deviant song, it will only show non-deviant songs. The album is split.

**Expected behavior**
The app assigns song 3 to their respective artists, so that you can see it on both A and B's author page. All the different albums are merged into only one, not split depending on the artist. This way, there are no duplicates in the app. Only one album, with all the songs inside.

**Smartphone (please complete the following information):**
 - Device: realme 9
 - OS: realme UI
 - App Version: 1.0.14

**Additional context**
I have only tested this with .m4a files, I don't know if metadata works differently with other formats like .mp3. This is probably a metadata issue, where the app doesn't recognize commas in the artist field probably. I think it interprets multiple names as one big name, a whole new artist.

I've noticed that songs downloaded with spotube don't have spaces after commas to separate each artist name. Maybe that's the issue?

- [x] I will follow the issue format and will add `[BUG]` to the start of my issue title, and I acknowledge that if I don't follow the format, my issue might get closed without further explanation.

**Comment by k0nstruct** [comment]
Update : it's a metadata issue, and it only happens with poorly formatted albums. The app makes albums for each *album author*, another field besides the album name. If that album author changes (with some files, some not) then the album is split. So why not just group by album name?
The app just needs to group songs with the same *album name* together, instead of the same *album artist*.

**Comment by AkaneTan** [comment]
In my phone there’s four albums with the same name. That happens a lot if you have a lot of albums in your library.

**Comment by k0nstruct** [comment]
Duplicate albums sprout from poorly formatted songs, where the field <album author> changes per song. But most songs respect the standard of putting the album name for each song.

**Comment by AkaneTan** [comment]
And since you’ve mentioned your songs are poorly tagged, I’m closing this issue as it’s not a software side problem.

**Comment by nift4** [comment]
>  So why not just group by album name?

Sometimes, two different albums have same name because they are by different artist
